stackrox/stackrox
# ROX-19759: Add administration events route and table

## Description

1. Do steps and write instructions for ui/apps/platform/README.md: Add a route to frontend code
    * routePaths.ts
    * Body.tsx
    * NavigationSidebar.tsx

2. Write AdministrationEventsService.ts file with residue:
    * filter for count request
    * filter and pagination for list request
    * See https://github.com/stackrox/stackrox/blob/master/proto/api/v1/administration_events_service.proto

3. Add src/Containers/Administration/Events folder

4. Add events page with residue:
    * empty state with and without filter
    * polling and last updated with conditionally rendered reload button like Network Graph

5. Add events table and toolbar with residue:
    * table filters and pagination

## Checklist
- [x] Wrote integration tests

## Testing Performed


Before `yarn deploy` in ui, do the following:

```sh
export ROX_ADMINISTRATION_EVENTS=true
```

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js: 604 = 3764957 - 3764353
        total: 15066 = 10092870 - 10077804
    * `ls -al build/static/js/*.js | wc -l`
        1 = 83 - 82 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/administration-events
    * See requests in browser Network panel
    * See only page heading and table head row because no events

### Integration testing

Before `yarn cypress-open` in ui/apps/platform, do the following:

```sh
export CYPRESS_ROX_ADMINISTRATION_EVENTS=true
```

* administation/events/eventsTable.test.js

    displays table head cells
    ![cypress_eventsTable1](https://github.com/stackrox/stackrox/assets/11862657/ac5607ee-6cbc-4e35-9caf-b86ec67be37c)

    has link to event page
    ![cypress_eventsTable2](https://github.com/stackrox/stackrox/assets/11862657/1f37aba3-2577-48a9-9722-c8183fbaedd2)
